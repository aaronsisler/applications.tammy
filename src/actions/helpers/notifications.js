export const setNotifications = (notifications) => ({
    type: 'SET_NOTIFICATIONS',
    notifications,
});
