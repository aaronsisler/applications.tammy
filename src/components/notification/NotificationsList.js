import React from 'react';
import PropTypes from 'prop-types';
import NotificationsListItem from 'Notification/NotificationsListItem';

export default class NotificationsList extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        if (this.props.notifications.length === 0) {
            return (
                <div id="notifications_list" className="empty">
                    No available items
                </div>
            );
        }

        return (
            <div id="notifications_list">
                <div className="notifications_list__header">
                    <div className="notifications_list__title">
                        Notifications
                    </div>
                    <div className="notifications_list__action">
                        <button className="button">Mark all as Read</button>
                    </div>
                </div>
                <div className="notifications_list__content">
                    {
                        this.props.notifications.map((notification) =>
                            <NotificationsListItem
                                key={notification.notificationId}
                                {...notification}
                            />)
                    }
                </div>
            </div>
        );
    }
}

NotificationsList.propTypes = {
    notifications: PropTypes.array,
};
