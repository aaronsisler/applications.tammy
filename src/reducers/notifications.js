const notificationsReducerDefaultState = [];

const notificationsReducer = (state = notificationsReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_NOTIFICATIONS':
            return action.notifications;
        default:
            return state;
    }
};

export default notificationsReducer;

