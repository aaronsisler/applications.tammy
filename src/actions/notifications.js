import database from 'Firebase/firebase';
import { setNotifications } from 'Actions/helpers/notifications';

export const startAddNotification = () => (undefined, getState) => {
    const { uid: userId } = getState().auth;
    const notification = { hasBeenViewed: false, notificationMessage: 'I am a notification' };

    return database.ref(`notifications/${userId}`).push({ ...notification });
}

export const startStreamNotifications = () => (dispatch, getState) => {
    const { uid: userId } = getState().auth;

    return database.ref(`notifications/${userId}`).on('value', (snapshot) => {
        const notifications = [];
        snapshot.forEach((childSnapshot) => {
            notifications.push({
                notificationId: childSnapshot.key,
                ...childSnapshot.val(),
            });
        });

        return dispatch(setNotifications(notifications));
    });
}
