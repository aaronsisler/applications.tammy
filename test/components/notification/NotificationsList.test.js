import React from 'react';
import { shallow } from 'enzyme';
import NotificationsList from 'Notification/NotificationsList';
import notifications from '../../fixtures/notifications';

describe('NotificationsList', () => {
    let wrapper;

    const buildWrapper = (notificationsInput = []) => {
        wrapper = shallow(
            <NotificationsList
                notifications={notificationsInput}
            />);
    }

    it('should render NotificationsList correctly when notifications are available', () => {
        buildWrapper(notifications);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render NotificationsList correctly when no notifications are available', () => {
        buildWrapper();
        expect(wrapper).toMatchSnapshot();
    });
});
