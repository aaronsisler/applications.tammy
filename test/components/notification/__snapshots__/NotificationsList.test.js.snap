// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsList should render NotificationsList correctly when no notifications are available 1`] = `
<div
  className="empty"
  id="notifications_list"
>
  No available items
</div>
`;

exports[`NotificationsList should render NotificationsList correctly when notifications are available 1`] = `
<div
  id="notifications_list"
>
  <div
    className="notifications_list__header"
  >
    <div
      className="notifications_list__title"
    >
      Notifications
    </div>
    <div
      className="notifications_list__action"
    >
      <button
        className="button"
      >
        Mark all as Read
      </button>
    </div>
  </div>
  <div
    className="notifications_list__content"
  >
    <Connect(NotificationsListItem)
      hasBeenViewed={false}
      key="firstNotificationId"
      notificationId="firstNotificationId"
      notificationMessage="First Notification"
    />
    <Connect(NotificationsListItem)
      hasBeenViewed={true}
      key="secondNotificationId"
      notificationId="secondNotificationId"
      notificationMessage="Second Notification"
    />
  </div>
</div>
`;
